### PR TITLE
hardtanh prelu softmax, test=develop

### PR DESCRIPTION
**Pull request by qili93**
Softmax
- paddle.nn.Softmax(axis=-1, name=None) - 添加此 class
- paddle.nn.functional.softmax(x, axis=-1, dtype=None, name=None - 完善文档和demo；input->x；添加dtype参数；当axis=-1 的时候，如果cudnn可用，就use_cudnn=True，其他情况下 use_cudnn=False；忽略 _stacklevel

PRelu
- paddle.nn.PReLU(num_parameters=1, init=0.25, weight_attr=None, name=None) - 添加此class, num_parameter决定是scalar还是channel模式，不再支持elementwise
- paddle.nn.functional.prelu(x, weight, name=None) - mode 保留, 支持float64

Hardtanh
- paddle.nn.Hardtanh(min=-1.0, max=1.0, name=None) - 添加此class
- paddle.nn.functional.hardtanh(x, min=-1, max=1, name=None) - 新增此function

class_Hardtanh
function_hardtanh
class_PRelu
function_prelu
class_Softmax
function_softmax

### CODE REVIEW

**Comment by qili93** [comment]
**Hardtanh Eng Doc**


class_Hardtanh
function_hardtanh

**Comment by qili93** [comment]
**PReLU Eng Doc**

class_PReLU
function_prelu

**Comment by qili93** [comment]
**Softmax Eng Doc**

class_Softmax
function_softmax

**Review by zhupengyang — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM
